src/scenic/simulators/newtonian/simulator.py

```diff
@@ -196,7 +196,22 @@ def isOnScreen(self, x, y):
     def step(self):
         for obj in self.objects:
             current_speed = obj.velocity.norm()
-            if hasattr(obj, "hand_brake"):
+            # 1) Pedestrian: we stash .control['heading']/['speed']
+            if hasattr(obj, "control") and "speed" in obj.control:
+                h = (
+                    obj.control["heading"]
+                    if obj.control["heading"] is not None
+                    else obj.heading
+                )
+                s = (
+                    obj.control["speed"]
+                    if obj.control["speed"] is not None
+                    else obj.speed
+                )
+                vel = Vector(0, s).rotatedBy(h)
+                obj.setVelocity(vel)
```

Comment:
```suggestion
                obj.velocity = Vector(0, s).rotatedBy(h)
```
I think this is easier to understand, since you don't need to go look up what `setVelocity` does.